src/cata_tiles.cpp

```diff
@@ -2757,6 +2758,20 @@ bool cata_tiles::draw_zone_mark( const tripoint &p, lit_level ll, int &height_3d
     return false;
 }
 
+bool cata_tiles::draw_zombie_revival_indicators( const tripoint &pos, const lit_level ll,
+        int &/*height_3d*/, const bool ( &/*invisible*/ )[5] )
```

Comment:
Please move these code to `draw_field_or_item` and handle overrides and visibility correctly (the revival indicators of the original items shouldn't show if a tile has an item override or the player cannot see the tile. `ll != LL_BLANK` alone wouldn't work).

Reply:
I did it in `draw_field_or_item` originally, but this resulted in vehicle parts being drawn on top of indicators making them invisible. I'll see if I'll be able to change some `draw_from_id_string` so it's always drawn on top, so I can move it back to `draw_field_or_item`.

I don't think revival indicators should be hidden behind some items when the player can see the tile, especially when you can see corpses with `V` menu.

Reply:
I don't think there's such a parameter in `draw_from_id_string`. AFAIK the sprites are drawn without depth information. I don't know what other people think about it, but if you really want to display the indicators this way, you can probably save the indicator positions in `draw_field_or_item` and draw them later in `draw_zombie_revival_indicators`.

Reply:
I could not come up with a solution where splitting the logic between `draw_field_or_item` and `draw_zombie_revival_indicators` would result in better performance/readability. It's just shifting code around + I'll have to store cached indicator positions somewhere. So, for now, I'll stick to a separate logic version.

The information about `draw_from_id_string` is very useful, thanks!

Reply:
However there's actually a problem with the current code: functions like `draw_field_or_item` are not only called when a tile is visible to the player: it can be called when a tile is not visible, but has been memorized or has override graphics set for the debug preview. Additionally a visible tile may contain items which are not visible to the player (for example, items inside lockers). Your current code does not check these conditions and will display the indicator over memory tiles, invisible items stacks, etc.

And since the logic to check whether a stack of items is visible is complicated, moving the logic, or part of it, to `draw_field_or_item` actually improves consistency.

Reply:
Can you please tell me more about overriding graphics set for the debug preview? How do I enable it? I made the necessary changes to make it play nice with other things you've listed.

Reply:
You need to set the keybindings for the debug menu if you haven't yet (press `?` from the main game UI, search for debug menu and set a key). Then open the debug menu and press `m`, `M`, and `o` (if I remember correctly), then select an mapgen entry from the menu. The mapgen contents will be shown in a blinking preview.